sourcerer-dbstore/src/main/java/org/elder/sourcerer2/dbstore/DbstoreEventRepository.kt

```diff
@@ -0,0 +1,359 @@
+package org.elder.sourcerer2.dbstore
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.yield
+import org.elder.sourcerer2.EventData
+import org.elder.sourcerer2.EventRecord
+import org.elder.sourcerer2.EventRepository
+import org.elder.sourcerer2.EventSubscriptionUpdate
+import org.elder.sourcerer2.ExpectedVersion
+import org.elder.sourcerer2.RepositoryReadResult
+import org.elder.sourcerer2.RepositoryVersion
+import org.elder.sourcerer2.StreamId
+import org.elder.sourcerer2.StreamReadResult
+import org.elder.sourcerer2.StreamVersion
+import org.elder.sourcerer2.exceptions.UnexpectedVersionException
+import org.slf4j.LoggerFactory
+
+internal class DbstoreEventRepository<T>(
+        private val repositoryInfo: DbstoreRepositoryInfo<T>,
+        private val eventStore: DbstoreEventStore,
+        private val objectMapper: ObjectMapper,
+        private val coroutineScope: CoroutineScope,
+        private val watchdogTimeoutMillis: Long = 10_000
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMagicNumber)